### PR TITLE
Poetry

### PR DESCRIPTION
**Pull request by patrit**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,10 +38,7 @@ Currently, we support BitBucket Server, GitHub and Gitlab.
 ### Setup
```
Comment (christiansiegel) [inline README.md:38]: Mention poetry version prerequisite. I tried it with the current [Gitpod config](https://github.com/baloise/gitopscli/blob/master/.gitpod.yml) (poetry v1.2.1) and it complains about the lock file format (v1.6.1). 
(We should update the Gitpod config in a follow-up. Seems like running the CLI locally in the devpod does not work due to some venv issue.)

**Review by christiansiegel — COMMENTED** [review]
Nice clean-up of the dependency mess!

Added some small comments. And I think https://github.com/baloise/gitopscli/blob/master/docs/setup.md#from-source needs to be updated or removed as well.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+pre-commit = "*"
+
+[tool.poetry.group.docs.dependencies]
+mkdocs = "*"
+mkdocs-material = "*"
+markdown-include = "*"
+pymdown-extensions = "*"
+Markdown = "*"
+
+[tool.ruff]
```
[the remark this answers is not in the document]
Comment (patrit) [reply pyproject.toml:37]: nope. maybe a little bit too preemptive - depends what people use. remove atm?

File: README.md (hunk 1 above)
Comment (patrit) [reply README.md:38]: was out of my focus. I will have a look ...

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (christiansiegel) [reply pyproject.toml:37]: Yeah, maybe remove since it's not used in the Makefile. But feel free to replace pylint with ruff in a follow-up PR :+1:

**Comment by patrit** [comment]
I think the setup documentation with pip still works as expected and according to the previous behavior.
`docker run --rm -t python:3.11 bash -c "git clone https://github.com/baloise/gitopscli.git && pip3 install gitopscli/ && gitopscli --help && pip3 uninstall gitopscli -y"`
What change did you expect?
Poetry does all the magic in virtualenvs and not in your local installation.
Should we change to virtualenvs for the proposed setup from source?

**Review by christiansiegel — APPROVED** [review]
lgtm :+1:

**Pull request merged**